[Term Entry] General / Semantic Technologies / RDF (#3913)
Codecademy/docs: content/general/concepts/semantic-technologies/terms/rdf.md

---
Title: 'rdf'
Description: 'A language for representing semantic data.'
Subjects:
  - 'Data Science'
Tags:
  - 'Data'
  - 'Semantic
  - 'Metadata'
  - 'Tuples'
CatalogContent:
  - 'paths/data-science-foundations'
  - 'paths/data-science-nlp'
---

**Resource Description Framework (RDF)** is a language for representing information. All data within the suite of semantic technologies is represented in RDF.

## RDF Triples

> RDF represents data in the format of **triples** -- ordered sets of three elements. The elements in an RDF triple stand for a subject, a predicate, and an object. An RDF triple can therefore be represented as follows:

Reviewer: ```suggestion
RDF represents data in the format of **triples** -- ordered sets of three elements. These elements in an RDF triple signify a subject, a predicate, and an object.
An RDF triple can therefore be represented as follows:
```

```pseudo
subject, predicate, object
```

- *`subject`*: The element which the triple says something about.
- *`predicate`*: A feature or property that applies to `subject`.
- *`object`*: The value of the feature or property represented by `predicate`.

> To illustrate, take the sentence 'The novel *Pachinko* was written by Min Jin Lee'. The sentence conveys something about *Pachinko*: that it has a feature of having an author, and that this author is Min Jin Lee. This information can be represented by the following triple:

Reviewer: ```suggestion
As an illustration, consider the sentence 'The novel Pachinko was written by Min Jin Lee.' This sentence communicates a detail about Pachinko: specifically, that it possesses a characteristic of having an author, and that this author is Min Jin Lee. This information can be depicted using the following triple:
```

Author: I used italics for 'Pachinko' as it is a convention to put book titles in italics.

```pseudo
Pachinko, hasAuthor, Min Jin Lee
```

### Triple Terms

In contrast to the above examples, the elements in a valid RDF triple cannot just be any values at all, such as arbitrary strings (e.g., `hasAuthor`). Rather, each of these elements can only be particular values:

- A `subject` is either an **International Resource Identifier (IRI)**, a **blank node**, or a **quoted triple**.
- A `predicate` must be an IRI.
- An `object` is either an IRI, a blank node, a quoted triple, or a **literal**.

> Each of these terms is explained below:

Reviewer: Please add an example like you initially added while introducing triple.

Author: @vaibhavmani810 Are you asking for examples of these terms ('IRI', 'blank node', etc.), or an example of a triple using these kinds of terms?

Reviewer: Example of a Triple using these kind of terms.

- *IRI*: A generalization of [URI](https://www.codecademy.com/resources/docs/general/uri), IRIs identify and link to resources on private and public networks.
- *Quoted triple*: An RDF triple used within a different RDF triple. Quoted triples allow statements to be made within RDF about other RDF statements.
- *Literal*: A value representing a number, string, boolean, date/time, or other simple values.
- *Blank node*: Blank nodes do not specify anything in particular. They allow for statements that say that something stands in a given relationship without identifying what that thing is. In practice, blank nodes are commonly avoided.

## RDF Serializations

Valid RDF must be a written in a concrete RDF syntax, also known as an **RDF serialization**. Two important RDF serializations are briefly described below.

### Turtle

**Turtle** is a compact RDF syntax intended to make RDF simple for humans to read and write.

> The most basic way to write RDF statements in Turtle is to enclose IRIs in angle brackets and end each statement with a period, like so:

Reviewer: ```suggestion
The most basic way to write RDF statements in Turtle is to enclose IRIs in angle brackets and end each statement with a period, as shown below:
```

```pseudo
<http://books.com/books#pachinko> <http://document-ontology.org/ontology#hasAuthor> <https://www.minjinlee.com/about> .
```

The above example represents the triple about *Pachinko* using IRIs that correspond to the subject (*Pachinko*), predicate (hasAuthor), and object (Min Jin Lee).

> However, Turtle allows for simpler formatting of statements by using IRI prefixes and grouping predicates and objects under a common subject, like so:

Reviewer: ```suggestion
However, Turtle allows for simpler formatting of statements by using IRI prefixes and grouping predicates and objects under a common subject, as shown below:
```
